plugins/out_es/es_conf.c

```diff
@@ -153,8 +157,41 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
             flb_es_conf_destroy(ctx);
             return NULL;
         }
+        flb_plg_debug(ctx->ins, "extracted cloud_host: '%s'", cloud_host);
+
+        flb_plg_debug(ctx->ins,
+                      "check whether extracted cloud_host '%s' contains a port",
+                      cloud_host);
+        cloud_host = strtok_r(cloud_host, ":", &saveptr);
+        flb_plg_debug(ctx->ins, "cloud_host without port: '%s'", cloud_host);
+        if (cloud_host == NULL) {
+            flb_plg_error(ctx->ins, "cannot extract cloud_host");
+            flb_es_conf_destroy(ctx);
+            return NULL;
+        }
+
+        cloud_port_char = strtok_r(NULL, ":", &saveptr);
```

Comment:
I'm not entirely sure about the current `strtok` situation but If I recall correctly we wanted it abstracted. In this particular case I think using `strtok` shouldn't be necessary since we could do the same using `strchr`, verifying the result and manually inserting the terminator character.

Reply:
It is about extracting the port not about inserting the `:`.

Reply:
Do you mean that strtok should not be used at all in the PR?

Reply:
When I said terminator I meant the NULL byte that would replace the colon.

Reply:
This is what I meant : 

```
        cloud_host = extract_cloud_host(ctx, tmp);

        if (cloud_host == NULL) {
            flb_plg_error(ctx->ins, "cannot extract cloud_host");
            flb_es_conf_destroy(ctx);
            
            return NULL;
        }

        flb_plg_debug(ctx->ins, "extracted cloud_host: '%s'", cloud_host);

        flb_plg_debug(ctx->ins,
                      "check whether extracted cloud_host '%s' contains a port",
                      cloud_host);

        cloud_port_char = strchr(cloud_host, ':');

        if (cloud_port_char == NULL) {
            flb_plg_debug(ctx->ins, "cloud_host without port: '%s'", cloud_host);
        }
        else {
            cloud_port_char[1] = '\0';
            cloud_port_char = &cloud_port_char[1];

            flb_plg_debug(ctx->ins, "extracted cloud_port_char: '%s'", cloud_port_char);

            cloud_host_port = (int) strtol(cloud_port_char, (char **) NULL, 10);

            flb_plg_debug(ctx->ins, "converted cloud_port_char to port int: '%i'", cloud_host_port);
        }

        if (cloud_host_port == 0) {
            cloud_host_port = cloud_port;
        }
```

Reply:
Applied strchr. Could you review again please?

Reply:
May I ask why did you deviate from this part? 

```
            cloud_port_char[1] = '\0';
            cloud_port_char = &cloud_port_char[1];
```

Your version should work as well but I don't see the reason to add one more variable or to delay the `\0` injection. What I do see is that I had an error in the sample code and it should be `cloud_port_char[0] = '\0';` instead of `cloud_port_char[1] = '\0';`, otherwise you're not replacing the colon in addition to zeroing the first digit of the port number which will cause the `strtol` call to return `0`.

If you could change that I'd really appreciate as it's easier to interpret at first glance.

Reply:
Updated the PR. Could you review again please?